pkg/test/fakes/gce_metadata_server/main.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	addr = ":8080"
+
+	projID     = "test-project"
+	projNumber = "123456789"
+	instance   = "test-instance"
+	instID     = "987654321"
+
+	metaPrefix     = "/computeMetadata/v1"
+	projIDPath     = metaPrefix + "/project/project-id"
+	projNumberPath = metaPrefix + "/project/numeric-project-id"
+	instIDPath     = metaPrefix + "/instance/id"
+	instancePath   = metaPrefix + "/instance/name"
+	attrKey        = "attribute"
+	attrPath       = metaPrefix + "/instance/attributes/{" + attrKey + "}"
+)
+
+var (
+	hostHeaders = []string{"metadata", "metadata.google.internal", "169.254.169.254"}
+
+	instAttrs = map[string]string{
```

Comment:
Will we want this configurable? Possible read from env var?

Reply:
Possibly. This was a first iteration at just getting something up to start building other pieces with.